### PR TITLE
Add file system vector-IO operations interface.

### PR DESCRIPTION
**Pull request by HHoflittlefish777**
Add file system Vector-IO operations interface.

### CODE REVIEW

File: pkg/fs/file_system.go [hunk 1]
```diff
@@ -44,10 +44,14 @@ type Dir interface {
 type File interface {
 	// Append mode, which adds new data to the end of a file.
 	AppendWriteFile(buffer []byte) error
+	// Vector Append mode, which supports appending consecutive buffers to the end of the file.
```
Comment (hanahmily) [inline pkg/fs/file_system.go:47]: What's the different from

```go
for _, b range buffers {
        f.AppendWriteFile(b)
}
    
```

Do you implement any optimizations for this operation?

File: docs/concept/persistence-storage.md [hunk 2]
```diff
@@ -100,7 +100,7 @@ return: File pointer, you can use it for various operations.
 
 ### Write
 BanyanDB provides two methods for writing files.
-Append mode, which adds new data to the end of a file. This mode is typically used for WAL.
+Append mode, which adds new data to the end of a file. This mode is typically used for WAL. And BanyanDB supports vector Append mode, which supports appending consecutive buffers to the end of the file.
```
Comment (hanahmily) [inline docs/concept/persistence-storage.md:103]: All append operations are "vector" style. Can you elaborate why we should highlight it.

**Review by hanahmily — CHANGES_REQUESTED** [review]

File: pkg/fs/file_system.go (hunk 1 above)
Comment (HHoflittlefish777) [reply pkg/fs/file_system.go:47]: Quite different, the difference between one system call and multiple system calls.

File: docs/concept/persistence-storage.md (hunk 2 above)
Comment (HHoflittlefish777) [reply docs/concept/persistence-storage.md:103]: A 'vector' refers to a continuous set of memory buffers. By using the writev system call, multiple buffers (i.e. vectors) can be written in one operation.
In writev, it is necessary to provide a vector composed of iovec structures, each describing the address and length of a buffer to be written. The content of this vector is equivalent to an array, hence it is called a "write vector".

Comment (HHoflittlefish777) [reply docs/concept/persistence-storage.md:103]: That is what linux define.

File: pkg/fs/file_system.go (hunk 1 above)
Comment (HHoflittlefish777) [reply pkg/fs/file_system.go:47]: For example, we can use writev system call in Linux to implement it.

Comment (hanahmily) [reply pkg/fs/file_system.go:47]: To ensure any changes made in a function affect the original two-dimensional slice, it's important to use `*[][]byte` as the parameter.

File: docs/concept/persistence-storage.md [hunk 3]
```diff
@@ -148,6 +156,14 @@ buffer: The read length is the same as the buffer length.
 
 `File.ReadFile(offset int, buffer []byte) (error)`
 
+For vector reading:
+
+param:
+
+iov: Discontinuous buffers in memory.
+
+`File.ReadvFile(iov [][]byte) (error)`
```
Comment (hanahmily) [inline docs/concept/persistence-storage.md:165]: As I mentioned, iov should be a pointer

**Review by hanahmily — APPROVED** [review]

**Comment by wu-sheng** [comment]
I can't see where we link to `docs/concept/persistence-storage.md` doc, how a user could read this?

File: docs/concept/persistence-storage.md [hunk 4]
```diff
@@ -112,7 +112,7 @@ return: File pointer, you can use it for various operations.
 
 ### Write
 BanyanDB provides two methods for writing files.
-Append mode, which adds new data to the end of a file. This mode is typically used for WAL.
+Append mode, which adds new data to the end of a file. This mode is typically used for WAL. And BanyanDB supports vector Append mode, which supports appending consecutive buffers to the end of the file.
```
[the remark this answers is not in the document]
Comment (HHoflittlefish777) [reply docs/concept/persistence-storage.md:116]: Capitalize just because it's the first letter. So, what should the specification be？

**Comment by hanahmily** [comment]
> I can't see where we link to `docs/concept/persistence-storage.md` doc, how a user could read this?

We have not implemented it. As WAL, once the implementation gets merged, we would publish this doc.

**Pull request merged**